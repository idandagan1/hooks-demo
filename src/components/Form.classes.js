import React from 'react';
import { ThemeContext } from '../context/theme';

export class Form extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      firstName: 'Idan',
      surname: 'Dagan',
      width: 0,
    };

    this.onSubmit = this.onSubmit.bind(this);
    this.onChange = this.onChange.bind(this);
    this.changeWidth = this.changeWidth.bind(this);
  }

  onChange(e) {
    const { firstName, lastName } = this.state;
    this.setState({ [e.target.name]: e.target.value });
    window.document.title = `${firstName} ${lastName}`;
  }

  componentDidMount() {
    window.addEventListener('resize', this.changeWidth);
  }

  componentWillUnmount() {
    window.removeEventListener('resize', this.changeWidth);
  }

  changeWidth() {
    this.setState({ width: window.innerWidth });
  }

  onSubmit(e) {
    e.preventDefault();
    const { firstName, surname } = this.state;
    console.log(`fistName: ${firstName}, surname: ${surname}`);
  }

  render() {
    const { firstName, lastName, width } = this.state;

    return (
      <div className="login-form">
        <div>Classes</div>
        <div>
          width:
          {width}
        </div>
        <ThemeContext.Consumer>
          {
          theme => (
            <form onSubmit={this.onSubmit}>
              <input
                value={firstName}
                onChange={this.onChange}
                name="firstName"
                placeholder="Your first name"
                autoComplete="off"
              />
              <input
                value={lastName}
                onChange={this.onChange}
                name="surname"
                placeholder="Your surname"
                autoComplete="off"
              />
              <button
                style={{ backgroundColor: theme }}
                type="submit"
              >
                login
              </button>
            </form>
          )
        }
        </ThemeContext.Consumer>
      </div>
    );
  }
}

export default Form;
