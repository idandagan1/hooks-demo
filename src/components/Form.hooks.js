import React, { useContext } from 'react';
import { ThemeContext } from '../context/theme';
import { useInput, useWindowTitle, useWindowWidth } from '../hooks';

export default function Form() {
  const firstName = useInput('Idan');
  const lastName = useInput('Dagan');
  const theme = useContext(ThemeContext);
  useWindowTitle(`${firstName.value} ${lastName.value}`, true);
  const width = useWindowWidth();

  function onSubmit() {
    console.log(`fistName: ${firstName}, surname: ${lastName}`);
  }

  return (
    <div className="login-form">
      <div>Hooks</div>
      <div>
        width:
        {width}
      </div>
      <form onSubmit={onSubmit}>
        <input
          {...firstName}
          placeholder="Your first name"
          autoComplete="off"
        />
        <input
          {...lastName}
          placeholder="Your last name"
          autoComplete="off"
        />
        <button
          style={{ backgroundColor: theme }}
          type="submit"
        >
          login
        </button>
      </form>

    </div>
  );
}
