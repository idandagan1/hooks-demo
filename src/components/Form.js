import React from 'react';
import { ThemeContext } from '../context/theme';

export class Form extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      firstName: '',
      surname: '',
    };

    this.onSubmit = this.onSubmit.bind(this);
    this.onChange = this.onChange.bind(this);
  }

  onChange(e) {
    this.setState({ [e.target.name]: e.target.value });
  }

  onSubmit(e) {
    e.preventDefault();
    const { firstName, surname } = this.state;
    console.log(`fistName: ${firstName}, surname: ${surname}`);
  }

  render() {
    const { firstName, lastName } = this.state;

    return (
      <div className="login-form">
        <ThemeContext.Consumer>
          {
          theme => (
            <form onSubmit={this.onSubmit}>
              <input
                value={firstName}
                onChange={this.onChange}
                name="firstName"
                placeholder="Your first name"
                autoComplete="off"
              />
              <input
                value={lastName}
                onChange={this.onChange}
                name="surname"
                placeholder="Your surname"
                autoComplete="off"
              />
              <button
                style={{ backgroundColor: theme }}
                type="submit"
              >
                login
              </button>
            </form>
          )
        }
        </ThemeContext.Consumer>
      </div>
    );
  }
}

export default Form;
