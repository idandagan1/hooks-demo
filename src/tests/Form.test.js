/* globals describe, it, expect, beforeAll */

import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';

import { Form } from '../components/Form';

configure({ adapter: new Adapter() });

describe('<Form />', () => {
  let wrapper;

  beforeAll(() => {
    wrapper = shallow(<Form />);
  });
  it('matches snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
