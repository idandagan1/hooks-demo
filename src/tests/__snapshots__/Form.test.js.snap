// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> matches snapshot 1`] = `
<div
  className="login-form"
>
  <ContextConsumer>
    <Component />
  </ContextConsumer>
</div>
`;
